src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -370,7 +370,53 @@ def deleteBlocks(self):
           If no => do nothing (check again next cycle)
         """
         # FIXME: figure out the proper logic for rule block deletion
+        # TODO: Update metadata after deleting block rules
         logging.info("Starting deleteBlocks methods --> IMPLEMENT-ME!!!")
+        logging.info("Starting deleteBlocks methods --> I'M ON IT...")
+
+        #Get list of blocks that can be deleted
+        blockDict = self.findDeletableBlocks.execute(transaction=False)
+
+        if not blockDict:
+            logging.info("No blocks were found")
+            return
+
+        deletableBlocks = []
+		# Obtain list of target RSE for each block
+        for blockName in blockDict:
+            rules = self.rucio.getParentContainerRules(name=blockName, scope=self.scope, account=self.rucioAcct)
+
+            #
+            requestedRSEs = []
+            requestedCopies = 0
+            for rule in rules:
+                if rule['account'] == self.rucioAcct:
+                    requestedCopies += rule['copies']
+                    requestedRSEs.extend(self.rucio.evaluateRSEExpression(rule['rse_expression'], useCache=True, returnTape=True))
+
+            #Get RSE in which the block is available
+            availableRSEs = self.rucio.getReplicaInfoForBlocks(block=blockName)
```

Comment:
If you map blocks to a container, as suggested in my comment above, you could then pass a list of those blocks (within the same container) to this method. Such that it makes a single HTTP requests for a bunch of blocks.
IF this chunk of blocks might become too big though (> 200 blocks), then we might need to apply some slicing on the list.